opentracing-cdi/src/main/java/io/opentracing/contrib/cdi/internal/SpanContextProducer.java

```diff
@@ -0,0 +1,47 @@
+package io.opentracing.contrib.cdi.internal;
+
+import io.opentracing.ActiveSpan;
+import io.opentracing.SpanContext;
+import io.opentracing.Tracer;
+import io.opentracing.contrib.web.servlet.filter.TracingFilter;
+
+import javax.enterprise.inject.Produces;
+import javax.enterprise.inject.spi.InjectionPoint;
+import javax.inject.Inject;
+import javax.servlet.http.HttpServletRequest;
+import java.util.logging.Logger;
+
+/**
+ * Produces a {@link SpanContext}, based on the {@link HttpServletRequest}'s attributes.
+ */
+public class SpanContextProducer {
```

Comment:
See:
https://github.com/jpkrohling/java-cdi/blob/5ca652c63a9e29c01f4118003265ca4705119b03/README.md#injecting-a-spancontext

and

https://github.com/opentracing-contrib/java-cdi/pull/1#discussion_r130870751